jsaddle/src/Language/Javascript/JSaddle/Run.hs

```diff
@@ -82,12 +83,25 @@ type CallbackResultId = ValId
 type CallbackResult = JSVal
 
 runJavaScript
-  :: (TryReq -> IO ()) -- ^ Send a request to the JS engine; we assume that requests are performed in the order they are sent; requests received while in a synchronous block must not be processed until the synchronous block ends (i.e. until the JS side receives the final value yielded back from the synchronous block)
+  :: ([TryReq] -> IO ()) -- ^ Send a batch of requests to the JS engine; we assume that requests are performed in the order they are sent; requests received while in a synchronous block must not be processed until the synchronous block ends (i.e. until the JS side receives the final value yielded back from the synchronous block)
   -> IO ( Rsp -> IO () -- Responses must be able to continue coming in as a sync block runs, or else the caller must be careful to ensure that sync blocks are only run after all outstanding responses have been processed
         , SyncCommand -> IO [Either CallbackResultId TryReq]
         , JSContextRef
         )
-runJavaScript sendReqAsync = do
+runJavaScript = runJavaScriptInt (500 {- 0.5 ms -}) 100
```

Comment:
Are these params just guesses, or was there some way we determined their value.

Reply:
Maybe leave a comment if it's the latter.

Reply:
I would bet we would get different results for different JSaddle backends, but 0.5 ms sounds like a reasonable default.

Reply:
Since master jsaddle does this, I thought there would be a similar hardcoded timeout like this, but couldn't find it. Am I missing it?

Reply:
The jsaddle master uses `TChan`. I cant recall why `TVar []` was used in this implementation. It makes sense to actually try `TChan` to see if that makes any difference in performance. There could be two outcomes, if there is a performance difference then we know which one is better, but if there isn't any difference then `TChan` is arguably a simpler implementation and should be preferred.

Reply:
@matthewbauer I reviewed the master branch code and here is my opinion. It loops over `TChan` till it becomes empty, or hits a 'sync' command (like inspecting the value of JSVal).
On the core-2 this approach will not work, as there are no 'sync' commands sent when inspecting the JSVal, rather it waits on the takeMVar (as the "value" of JSVal is sent automatically). So even if we use the TChan, we will have to signal immediate send of batch before stopping on the takeMVar, and the implementation will not be any better than the current using `TVar []` and `sendReqsBatchVar`

Reply:
I will add comment how these values were determined.